client/components/AddTestToQueueWithConfirmation/index.jsx

```diff
@@ -23,13 +26,30 @@ function AddTestToQueueWithConfirmation({
     const [showConfirmation, setShowConfirmation] = useState(false);
     const [addTestPlanReport] = useMutation(ADD_TEST_QUEUE_MUTATION);
     const [scheduleCollection] = useMutation(SCHEDULE_COLLECTION_JOB_MUTATION);
+    const { data: testPlanReportsInitiatedByAutomation } = useQuery(
+        TEST_PLAN_RUN_REPORTS_INITIATED_BY_AUTOMATION,
+        {
+            variables: {
+                testPlanVersionId: testPlanVersion?.id
+            },
+            fetchPolicy: 'cache-and-network',
+            skip: !testPlanVersion?.id
+        }
+    );
+
     const { triggerLoad, loadingMessage } = useTriggerLoad();
     const buttonRef = useRef();
 
     const hasAutomationSupport = isSupportedByResponseCollector({
         at,
         browser
     });
+    const alreadyHasBotInTestPlanReport =
+        testPlanReportsInitiatedByAutomation?.testPlanVersion.testPlanReports.some(
```

Comment:
looking back at this now. I would like to change this to `useMemo`. I will do this later today.